R/rank_becas.R

```diff
@@ -0,0 +1,298 @@
+        dplyr::mutate(r2_ratio_unnorm = R2m_unnorm/R2c_unnorm)
+      
+      # combat version
+      ball_nest <- omicsData_bc$e_data %>%
+        tidyr::pivot_longer(-dplyr::all_of(edata_cname),
+                            names_to = fdata_cname,
+                            values_to = "abundance") %>%
+        dplyr::left_join(omicsData_bc$f_data, by = fdata_cname) %>%
+        dplyr::group_by(!!as.symbol(edata_cname)) %>%
+        tidyr::nest()
+      
+      ball_nest <- ball_nest %>%
+        dplyr::mutate(glmm_r2 = purrr::map(data,function(dat){
+          lmer_res = lme4::lmer(abundance ~ !!as.symbol(main_effect_cname) + (1|!!as.symbol(batch_effect_cname)), data = dat,
+                                control = lme4::lmerControl(check.conv.singular = lme4::.makeCC(action = "ignore", tol = 1e-4)))
+          MuMIn::r.squaredGLMM(lmer_res)
+        }))
+      
+      ball_r2_bc <- ball_nest %>% 
+        dplyr::mutate(R2m = sapply(glmm_r2,'[',1),
+                      R2c = sapply(glmm_r2,'[',2)) %>%
+        dplyr::select(!!as.symbol(edata_cname),R2m,R2c) %>%
+        dplyr::mutate(r2_ratio = R2m/R2c)
+      
+      ball_all <- ball_r2_bc %>%
+        dplyr::left_join(ball_r2_unnormalized,by = edata_cname)
+      
+      abund_df = omicsData_unnormalized$e_data %>%
+        tibble::remove_rownames() %>%
+        tibble::column_to_rownames(var = edata_cname) %>%
+        as.matrix() %>%
+        matrixStats::rowMedians(na.rm=T) %>%
+        data.frame() %>%
+        tibble::rownames_to_column(var = edata_cname) %>%
+        dplyr::rename("medAbund" = ".")
+      
+      quartile_info = quantile(abund_df$medAbund)
+      abund_df <- abund_df %>%
+        dplyr::mutate(quantileNum = ifelse(medAbund < quartile_info[2],"Q1",
+                                           ifelse(medAbund > quartile_info[4],"Q4",
+                                                  ifelse(medAbund > quartile_info[2] & medAbund < 
+                                                           quartile_info[3],"Q2","Q3")))) %>%
+        dplyr::filter(quantileNum %in% c("Q3","Q4"))
+      
+      abundant_molecules = abund_df[,which(colnames(abund_df) == edata_cname)]
+      
+      ball_all_med <- ball_all %>%
+        dplyr::filter(!!as.symbol(edata_cname) %in% abundant_molecules) %>%
+        dplyr::mutate(r2_ratio_diff = r2_ratio - r2_ratio_unnorm) %>%
+        dplyr::ungroup() %>%
+        dplyr::summarise(medDiff = median(r2_ratio_diff))
+      
+      ball_all_med$medDiff
+    })
+    ranking_df = data.frame(r2_scores) %>% t() %>% data.frame() %>%
+      dplyr::rename(Value = ".") %>%
+      dplyr::arrange(desc(Value)) %>%
+      dplyr::mutate(Ranking = seq(from = 1, to = length(omicsData_beca_list), by = 1)) %>%
+      tibble::rownames_to_column(var = "BECA")
+    # update for ties
+    if(length(ranking_df$Value) != length(unique(ranking_df$Value))){
+      for(i in 2:nrow(ranking_df)){
+        if(ranking_df$Value[i] == ranking_df$Value[i-1]){
```

Comment:
I got the following error:
```
Error in if (ranking_df$Value[i] == ranking_df$Value[i - 1]) { : 
  missing value where TRUE/FALSE needed
```

I'll see if I can look into it more a bit later

Reply:
Yeah, it works on my end with this branch to create all the vignettes, so let me know if there is something I need to fix on my end

Reply:
What version of MuMIn are you using? Are you getting a warning saying `'r.squaredGLMM' now calculates a revised statistic`?

Reply:
So it looks like what's happening for me is on line 275, the values in the Molecule column are Var1, Var2, Var3, etc. and the values in abundant_molecules are 1, 2, 3, etc., so the filter removes all rows, meaning the median is NA which breaks stuff further down.

Reply:
Probably not a good solution long-term, but replacing lines 274-278 with
```
ball_all_med <- ball_all %>%
        dplyr::mutate(Molecule = gsub("Var", "", !!as.symbol(edata_cname))) %>%
        dplyr::filter(Molecule %in% abundant_molecules) %>%
        dplyr::mutate(r2_ratio_diff = r2_ratio - r2_ratio_unnorm) %>%
        dplyr::ungroup() %>%
        dplyr::summarise(medDiff = median(r2_ratio_diff))
```
Seemed to fix it for me.

Reply:
I have updated the code to suprress the warning message for the revised statistic since the revised statistic is what we want. Does that fix the vignette? In my code, the abundant_molecules are "Var15", "Var22", etc. which matches what I expected from the Molecule name column so your code update ends up breaking on my end. If that does not, we might need to create a meeting to go line by line and see why our abundant_molecules vector is different.

Reply:
Okay, so actually now that I know that it should have "Var" on the front I think I figured out the issue.
On line 259 the `matrixStats::rowMedians` call is getting rid of the names, so they're just turning into numbers. Replacing it with `matrixStats::rowMedians(na.rm=T, useNames = T)` seems to fix it.

Reply:
Great, I have updated the code accordingly. Thanks for identifying that!

Reply:
It should be `useNames`, (not use_names), and line 248 in test_rank_becas.R also needs to be changed. After those I think it'll be good to merge.

Reply:
Gotcha, I fixed the name of the argument for the function and added that distinction into the test file as well. Thanks!